youtube_dl/extractor/roosterteeth.py

```diff
@@ -0,0 +1,303 @@
+        if data and 'season' in data:
+            res['title'] = '%s: %s' % (data['season'], res['raw_title'])
+            res['season'] = data['season']
+        else:
+            res['title'] = res['raw_title']
+
+        return res
+
+    def _login(self, domain='roosterteeth.com'):
+        """
+        Attempt to log in to RoosterTeeth (or Achievement Hunter).
+        NOTE: RT is planning to implement SSO which will probably change how this works.
+        """
+
+        if domain in self._authed:
+            return self._authed[domain]
+
+        (username, password) = self._get_login_info()
+
+        # No authentication to be performed
+        if username is None:
+            return False
+
+        LOGIN_URL = 'http://%s/login' % domain
+        login_page, hdl = self._download_webpage_handle(
+            LOGIN_URL, None,
+            note='Downloading login page',
+            errnote='unable to fetch login page', fatal=False)
+
+        if login_page is False:
+            return False
+
+        if hdl.geturl() != LOGIN_URL:
+            # We were redirected which means that we're already logged in.
+            self._authed[domain] = True
+            return True
+        
+        token = self._search_regex(r'(?s)<input.+?name="_token".+?value="(.+?)"',
+                                   login_page, 'Login token')
+
+        # Log in
+        login_form_strs = {
+            '_token': token,
+            'username': username,
+            'password': password
+        }
+
+        # Convert to UTF-8 *before* urlencode because Python 2.x's urlencode
+        # chokes on unicode
+        login_form = dict((k.encode('utf-8'), v.encode('utf-8')) for k, v in login_form_strs.items())
+        login_data = compat_urllib_parse.urlencode(login_form).encode('ascii')
+
+        req = compat_urllib_request.Request(LOGIN_URL, login_data, {'Content-Type': 'application/x-www-form-urlencoded'})
+        login_results = self._download_webpage(
+            req, None,
+            note='Logging in', errnote='unable to log in', fatal=False)
+        
+        if login_results is False:
+            return False
+
+        if login_results.find('Error in exception handler.') > -1 or login_results.find('Authentication failed. Please check and try again, or reset your password') > -1:
+            self.report_warning('unable to log in: bad username or password')
+            self._authed[domain] = False
+            return False
+
+        self._authed[domain] = True
+        return True
+
+    def _is_sponsor(self, domain='roosterteeth.com'):
+        if self._sponsor is None:
+            username, _ = self._get_login_info()
+            profile_page = 'http://%s/user/%s' % (domain, compat_urllib_parse.quote(username))
+            html = self._download_webpage(
+                profile_page, None,
+                note='Checking user profile...',
```

Comment:
I think we don't end messages with `...` in any other extractor, so for consistency I would just use `'Checking user profile'`.